virtual_rainforest/models/abiotic_simple/abiotic_simple_model.py

```diff
@@ -0,0 +1,232 @@
+exists in a
+more complete state, and unpacks a small number of parameters from our currently pretty
+minimal configuration dictionary. These parameters are then used to generate a class
+instance. If errors crop here when converting the information from the config dictionary
+to the required types they are caught and then logged, and at the end of the unpacking
+an error is thrown. This error should be caught and handled by downstream functions so
+that all model configuration failures can be reported as one.
+"""  # noqa: D205, D415
+
+from __future__ import annotations
+
+from typing import Any, Dict, List
+
+from pint import Quantity
+from xarray import DataArray
+
+from virtual_rainforest.core.base_model import BaseModel
+from virtual_rainforest.core.data import Data
+from virtual_rainforest.core.exceptions import InitialisationError
+from virtual_rainforest.core.logger import LOGGER
+
+# The simple_regression.py is called by self.setup and self.update, will follow soon
+# from virtual_rainforest.models.abiotic_simple import simple_regression
+
+
+class AbioticSimpleModel(BaseModel):
+    """A class describing the abiotic simple model.
+
+    Args:
+        data: The data object to be used in the model.
+        update_interval: Time to wait between updates of the model state.
+        soil_layers: The number of soil layers to be modelled.
+        canopy_layers: The initial number of canopy layers to be modelled.
+        initial_soil_moisture: The initial soil moisture for all layers.
+    """
+
+    model_name = "abiotic_simple"
+    """The model name for use in registering the model and logging."""
+    lower_bound_on_time_scale = "1 day"
+    """Shortest time scale that soil model can sensibly capture."""
+    upper_bound_on_time_scale = "30 day"
+    """Longest time scale that soil model can sensibly capture."""
+    required_init_vars = (  # TODO add temporal axis
+        ("air_temperature_ref", ("spatial",)),
+        ("relative_humidity_ref", ("spatial",)),
+        ("atmospheric_pressure_ref", ("spatial",)),
+        ("precipitation", ("spatial",)),
+        ("atmospheric_co2", ("spatial",)),
+        ("mean_annual_temperature", ("spatial",)),
+        ("leaf_area_index", ("spatial",)),
+        ("layer_heights", ("spatial",)),
+    )
+    """The required variables and axes for the abiotic simple model"""
+
+    def __init__(
+        self,
+        data: Data,
+        update_interval: Quantity,
+        soil_layers: int,
+        canopy_layers: int,
+        initial_soil_moisture: float,
+        **kwargs: Any,
+    ):
+        # sanity checks for soil and canopy layers
+        if soil_layers < 1:
+            to_raise = InitialisationError(
+                "There has to be at least one soil layer in the abiotic model!"
+            )
+            LOGGER.error(to_raise)
+            raise to_raise
+
+        if soil_layers != int(soil_layers):
```

Comment:
It might be better to test here that `soil_layers` is an `int` object? This allows floats with no fractional part through - and that seems functionally fine - but it is probably best to ensure that only actual `int` values make it here?

```suggestion
        if not isinstance(soil_layers, int):
```